File: Net/Zeromq/src/main/java/org/elkoserver/foundation/net/zmq/Thunk.java
package org.elkoserver.foundation.net.zmq;

/**
 * An arbitrary (zero-argument) executable.  Similar to {@link Runnable} but
 * can throw an exception.
 *
 * @see java.lang.Runnable
 */
public interface Thunk {
    /**
     * Execute this thunk.
     */
    void run() throws Throwable;
}

File: Net/Zeromq/src/main/kotlin/org/elkoserver/foundation/net/zmq/Thunk.kt
package org.elkoserver.foundation.net.zmq

/**
 * An arbitrary (zero-argument) executable.  Similar to [Runnable] but
 * can throw an exception.
 *
 * @see java.lang.Runnable
 */
interface Thunk {
    /**
     * Execute this thunk.
     */
    fun run()
}
